### PR TITLE
LegacyScriptPubKeyMan code cleanups

### PR DESCRIPTION
**Pull request by ryanofsky**
This PR implements suggested code cleanups from #17300 and #17304 review comments

### CODE REVIEW

File: src/wallet/rpcwallet.cpp [hunk 1]
```diff
@@ -3766,14 +3766,7 @@ UniValue getaddressinfo(const JSONRPCRequest& request)
     ScriptPubKeyMan* spk_man = pwallet->GetScriptPubKeyMan();
     if (spk_man) {
         CKeyID key_id = GetKeyForDestination(*provider, dest);
```
Comment (Sjors) [inline src/wallet/rpcwallet.cpp]: Maybe add a comment or assertion that only LegacyScriptPubKeyMan should use `key_id`.

**Review by Sjors — APPROVED** [review]
Code review ACK ~5ed6daf4b703b6811d7b9f366edba35042de31f7~ 5b3a0b4

**Comment by ryanofsky** [comment]
Updated 5ed6daf4b703b6811d7b9f366edba35042de31f7 -> 5b3a0b4fbebcb84c6ab83df40c00be075d5ef28b ([`pr/keyman-cleanup2.2`](https://github.com/ryanofsky/bitcoin/commits/pr/keyman-cleanup2.2) -> [`pr/keyman-cleanup2.3`](https://github.com/ryanofsky/bitcoin/commits/pr/keyman-cleanup2.3), [compare](https://github.com/ryanofsky/bitcoin/compare/pr/keyman-cleanup2.2..pr/keyman-cleanup2.3)) with requested code comment

**Comment by achow101** [comment]
I'm not sure that the `GetMetadata` change is really the right direction. What I was thinking was more of passing in the scriptPubKey directly (or maybe a CTxDestination?) and pulling from it the key id or script id via the Solver.

**Comment by ryanofsky** [comment]
Updated 5b3a0b4fbebcb84c6ab83df40c00be075d5ef28b -> 1f54dd2184cd41268c5c6739befaec4cd60219ae ([`pr/keyman-cleanup2.3`](https://github.com/ryanofsky/bitcoin/commits/pr/keyman-cleanup2.3) -> [`pr/keyman-cleanup2.4`](https://github.com/ryanofsky/bitcoin/commits/pr/keyman-cleanup2.4), [compare](https://github.com/ryanofsky/bitcoin/compare/pr/keyman-cleanup2.3..pr/keyman-cleanup2.4)) replacing `GetMetadata` `CScript` argument with `CTxDestination`

**Comment by achow101** [comment]
Code review ACK 1f54dd2184cd41268c5c6739befaec4cd60219ae

**Comment by Sjors** [comment]
AppVeyor [failure](https://ci.appveyor.com/project/DrahtBot/bitcoin/builds/28640009#L2433) (could be random, I restarted it):

```
      C:\projects\bitcoin\build_msvc\x64\Release\libleveldb\libleveldb.lib
2433         libzmq-mt-s-4_3_3.lib(zmq.cpp.obj) : MSIL .netmodule or module compiled with /GL found; restarting link with /LTCG; add /LTCG to the link command line to improve linker performance
2434    14>bech32_tests.obj : error LNK2001: unresolved external symbol "bool __cdecl CaseInsensitiveEqual(class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > const &,class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > const &)" (?CaseInsensitiveEqual@@YA_NAEBV?$basic_string@DU?$char_traits@D@std@@V?$allocator@D@2@@std@@0@Z) [C:\projects\bitcoin\build_msvc\test_bitcoin\test_bitcoin.vcxproj]
2435    14>C:\projects\bitcoin\build_msvc\x64\Release\test_bitcoin.exe : fatal error LNK1120: 1 unresolved externals [C:\projects\bitcoin\build_msvc\test_bitcoin\test_bitcoin.vcxproj]
2436    14>Done Building Project "C:\projects\bitcoin\build_msvc\test_bitcoin\test_bitcoin.vcxproj" (default targets) -- FAILED.
2437     1>Done Building Project "C:\projects\bitcoin\build_msvc\bitcoin.sln" (default targets) -- FAILED.
2438
2439Build FAILED.
2440
2441       "C:\projects\bitcoin\build_msvc\bitcoin.sln" (default target) (1) ->
2442       "C:\projects\bitcoin\build_msvc\test_bitcoin\test_bitcoin.vcxproj" (default target) (14) ->
2443       (Link target) -> 
2444         bech32_tests.obj : error LNK2001: unresolved external symbol "bool __cdecl CaseInsensitiveEqual(class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > const &,class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > const &)" (?CaseInsensitiveEqual@@YA_NAEBV?$basic_string@DU?$char_traits@D@std@@V?$allocator@D@2@@std@@0@Z) [C:\projects\bitcoin\build_msvc\test_bitcoin\test_bitcoin.vcxproj]
2445         C:\projects\bitcoin\build_msvc\x64\Release\test_bitcoin.exe : fatal error LNK1120: 1 unresolved externals [C:\projects\bitcoin\build_msvc\test_bitcoin\test_bitcoin.vcxproj]
2446
2447    0 Warning(s)
2448    2 Error(s)
```

**Comment by Sjors** [comment]
Code review ACK 1f54dd2

**Comment by ryanofsky** [comment]
> AppVeyor [failure](https://ci.appveyor.com/project/DrahtBot/bitcoin/builds/28640009#L2433) (could be random, I restarted it):

Appveyor error is apparently unrelated and fixed by https://github.com/bitcoin/bitcoin/pull/17384#issuecomment-550305021

**Comment by ryanofsky** [comment]
Updated 1f54dd2184cd41268c5c6739befaec4cd60219ae -> 05b224a175065aee4d6d9c471722bc4503f01fdf ([`pr/keyman-cleanup2.4`](https://github.com/ryanofsky/bitcoin/commits/pr/keyman-cleanup2.4) -> [`pr/keyman-cleanup2.5`](https://github.com/ryanofsky/bitcoin/commits/pr/keyman-cleanup2.5), [compare](https://github.com/ryanofsky/bitcoin/compare/pr/keyman-cleanup2.4..pr/keyman-cleanup2.5)) just fixing now outdated `GetMetadata` comment

**Comment by Sjors** [comment]
re-ACK 05b224a

**Comment by laanwj** [comment]
Code review ACK 05b224a175065aee4d6d9c471722bc4503f01fdf

**Pull request merged**